### PR TITLE
feat: Add support for op-deployer, fix multiple L2s

### PR DESCRIPTION
**Pull request by mslipper**
Depends on https://github.com/ethereum-optimism/optimism/pull/12033.

### CODE REVIEW

**Comment by barnabasbusa** [comment]
From a very quick look, it seems like "manual deployment" of the contract is no longer possible. 
I might presume incorrectly but don't you ever wanna check that the manual contract deployment still working as expected? I understand that you would default to a contract being part of the genesis file, but feel like if you disable the functionality on contract deployment altogether then it's also going to be one more thing to test in case there is a change on L1.

**Comment by mslipper** [comment]
Hey! I'm not sure what you mean by manual deployment of the contracts. The contracts are deployed using regular transactions to the L1 in the enclave, they're just all deployed upfront. They're not injected into the L1's genesis file.

File: src/wait/wait_for_sync.star [hunk 1]
```diff
@@ -19,3 +19,14 @@ def wait_for_sync(plan, l1_config_env_vars):
             if [ "$is_optimistic" == "false" ]; then echo \'Node is synced!\'; break; fi; done',
         wait="72h",
     )
+
+
+def wait_for_startup(plan, l1_config_env_vars):
+    plan.run_sh(
+        name="wait-for-l1-startup",
+        description="Wait for L1 to start up - can take up to 2 minutes",
+        image="mslipper/deployment-utils:latest",
```
[the remark this answers is not in the document]
Comment (mslipper) [reply src/wait/wait_for_sync.star:28]: I'm already using this image in a bunch of places, so I thought I'd default to this one here as well for consistency. It's useful if we ever want to use Foundry tooling to query the chain state.

File: src/package_io/sanity_check.star [hunk 2]
```diff
@@ -19,7 +20,6 @@ SUBCATEGORY_PARAMS = {
         "holocene_time_offset",
         "interop_time_offset",
     ],
-    "op_contract_deployer_params": ["image"],
```
[the remark this answers is not in the document]
Comment (mslipper) [reply src/package_io/sanity_check.star:22]: added the validation back

File: src/package_io/input_parser.star [hunk 3]
```diff
@@ -156,10 +175,32 @@ def default_participant():
...
         "image": "ethpandaops/optimism-contract-deployer:develop",
+        "artifacts_url": "https://storage.googleapis.com/oplabs-contract-artifacts/artifacts-v1-4accd01f0c35c26f24d2aa71aba898dd7e5085a2ce5daadc8a84b10caf113409.tar.gz",
+    }
+
+
+def default_ethereum_config():
+    return {
+        "network_params": {
+            "preset": "minimal",
+            "seconds_per_slot": 3,
```
[the remark this answers is not in the document]
Comment (mslipper) [reply src/package_io/input_parser.star]: fixed

File: src/wait/wait_for_sync.star (hunk 1 above)
[the remark this answers is not in the document]
Comment (barnabasbusa) [reply src/wait/wait_for_sync.star:28]: Sounds good, but then could we move it to a variable? If it's used in many places?

[the remark this answers is not in the document]
Comment (mslipper) [reply src/wait/wait_for_sync.star]: done!

**Review by barnabasbusa — APPROVED** [review]

**Pull request merged**